File: python/challenge14.py
from decimal import getcontext, Decimal
getcontext().prec = 3

def dynamicPricing(numberOfPeople, distanceTraveled):
  basePrice = 1.00 + distanceTraveled*0.25
  totalPrice = (Decimal(basePrice) + Decimal(0.25)) if (numberOfPeople >= 30) else Decimal(basePrice)
  return '$' + str(Decimal(totalPrice))

File: ruby/challenge14.rb
def dynamicPricing(numberOfPeople, distanceTraveled)
    basePrice = (1.00 + (distanceTraveled.to_f*0.25).round(3)).round(3)
    totalPrice = (numberOfPeople >= 30) ? basePrice.to_f.round(3) + 0.25 : basePrice.to_f.round(3)
    return '$' + totalPrice.round(3).to_s
end
  
# numberOfPeople = 30
# distanceTraveled = 10
# dynamicPricing(numberOfPeople, distanceTraveled)
